sbi/neural_nets/embedding_nets/transformer.py

```diff
@@ -628,91 +628,123 @@ def forward(self, pixel_values: torch.FloatTensor) -> torch.Tensor:
 
 
 class TransformerEmbedding(nn.Module):
-    def __init__(self, config):
+    def __init__(
+        self,
+        *,
+        pos_emb: str = "rotary",
+        pos_emb_base: float = 10e4,
+        rms_norm_eps: float = 1e-05,
+        router_jitter_noise: float = 0.0,
+        vit_dropout: float = 0.5,
+        mlp_activation: str = "gelu",
+        is_causal: bool = True,
+        vit: bool = False,
+        num_hidden_layers: int = 4,
+        num_attention_heads: int = 12,
+        num_key_value_heads: int = 12,
+        intermediate_size: int = 256,
+        ffn: str = "mlp",
+        head_dim: Optional[int] = None,
+        attention_dropout: float = 0.5,
+        feature_space_dim: int,
+        final_emb_dimension: Optional[int] = None,
+        image_size: Optional[int] = None,
+        patch_size: Optional[int] = None,
+        num_channels: Optional[int] = None,
+        num_local_experts: Optional[int] = None,
+        num_experts_per_tok: Optional[int] = None,
+    ):
         super().__init__()
         """
         Main class for constructing a transformer embedding
         Basic configuration parameters:
-            pos_emb (string): position encoding to be used, currently available:
+            pos_emb: position encoding to be used, currently available:
```

Comment:
if arg docstrings break the line please add a small indent to set them apart from other arguments.